### ISSUE TITLE
Instance Image Retrieval 평가 방법 정리

### ISSUE THREAD

**Issue by chullhwan-song** [chullhwan-song/Image-Retrieval#1]
@

**Comment by chullhwan-song** [comment]
# DIR case
## 학습셋
* we start with the publicly available models pretrained on the ImageNet ILSVRC data. 
* Weakly Data
   * **Neural codes** for image retrieval : http://sites.skoltech.ru/compvision/projects/neuralcodes/
        *  a large-scale image dataset that contains approximately 214k images of 672 famous landmark. sites. Its images were collected through textual queries in an image search engine without thorough verification.        
    * We obtained a set of about 192,000 images divided into 586 landmarks. We refer to this set as **Landmarks full**. For our experiments, we use 168,882 images for the actual finetuning, and the 20,668 remaining ones to validate parameters. 
       * > Neural codes에서 제공한 URL이 깨진것등에 의해 다운받을수 없는 문제들이 있어서..214k -> 192k로 줄었다는 의미
* Cleaning the Landmarks dataset : http://www.europe.naverlabs.com/Research/Computer-Vision/Learning-Visual-Representations/Deep-Image-Retrieval
   * SIFT같은 local descriptor등의 알고리즘들을 이용,  위의 weakly한 **Neural codes**를 clean하게 생성.
   *  This cleaning process leaves about 49,000 images (divided in 42,410 training and 6,382 validation images) still belonging to one of the 586 landmarks, referred to as **Landmarks-clean**. 
   * 현재 저 링크에 관련 다운로드가 안됨. ㅠㅠ (따로 받은게 있긴한데..)
* 테스트셋
    * Oxford 5k, Paris 6k, Holidays and  UKB datasets 
    * we also consider the Oxford 105k and Paris 106k datasets that extend Oxford 5k and Paris 6k with **100k distractor images**
* 평가방법
    * UKB 데이터셋 : recall@4 , 나머지 데이터셋은  mean average precision (mAP) 
    * As is standard practice, in Oxford and Paris one uses only the annotated region of interest of the query, while for Holidays and UKB one uses the whole query image. 
       * 관례대로 Oxford와 Paris : 쿼리의 주석이 달린 관심 영역(ROI)만 사용 
        * Holidays와 UKB:  전체 쿼리 이미지
    * the query image is removed from the dataset when evaluating on Holidays, but not on Oxford, Paris, and UKB.
         * Holidays 데이터셋 평가 :  쿼리 이미지가 데이터 집합에서 제거
         * Oxford 5k, Paris 6k,  UKB 데이터셋 평가 : 에서는 쿼리 이미지가 제거되지 않음
* 의문
   * Oxford 5k, Paris 6k, Holidays and  UKB datasets의 경우는 그중 몇개에서 DB에 넣고 평가하는건가?  **100k distractor images**를 집어넣지 않고 하면 쉬운 문제 아닌가??
* 소스
    * https://github.com/figitaki/deep-retrieval 
        * 학습 소스는 공개되지 않았음.
        * Caffe 기반.

**Comment by chullhwan-song** [comment]
# DELF 케이스 
* 학습셋
    * DIR 과 동일
    * train set
       * full version : 140,372 images from 586 landmarks
       * clean version : 35,382 images of 586 landmarks
   * We use LF to train our attention model, and LC is employed to fine-tune the network for image retrieval.
        * 먼저, clean version으로 하고, full version으로 두번째 한다는 의미 - softmax loss 둘다.
* 테스트셋
    * Oxf5k, Par6k, Oxf105k and Par106k, for completeness. 
* 평가방법
